students/Z_shen/lesson04/mailroom_2.py

```diff
@@ -0,0 +1,143 @@
+import operator
+donor = {'William Gates': [1500.99, 3500, 800.25],
+         'Jeff Bezos': [145.72, 1350.25],
+         'Paul Allen': [250.00, 57.00],
+         'Mark Zuckerberg': [600.00],
+         }
+
+
+def sort_donor_total_given(donor):
+    result = {}
+    new_donor_order = {}
+    for key, value in donor.items():
+        result.update({key: sum(value)})
+        sort = sorted(result.items(), key=operator.itemgetter(1), reverse=True)
+    for name, num in sort:
+        new_donor_order[name] = num
+    return new_donor_order
+
```

Comment:
calling sorted() inside of a loop is inefficient, you only need to call sorted() once.

here is what the simplified version looks like:
```

def sort_func(item):
    return sum(item[1])

sorted_data = sorted(donors.items(), key=sort_func, reverse=True)
```
from there you can loop over sorted_data and print the report info